src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -171,6 +171,259 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                 writer.block_end();
             }
 
+            template <>
+            void GPU_Emitter::EMITTER_DECL(ngraph::op::Convolution)
+            {
+                if (out[0].get_size() == 0)
+                {
+                    return;
+                }
+
+                const std::string args0 = "x_descriptor";
+                const std::string args1 = "w_descriptor";
+                const std::string out0 = "y_descriptor";
+                const std::string conv_descriptor = "conv_descriptor";
+                const std::string data_type = "CUDNN_DATA_FLOAT";
+                const std::string tensor_format = "CUDNN_TENSOR_NCHW";
+                const std::string mode = "CUDNN_CROSS_CORRELATION";
+                const std::string conv_algo = "CUDNN_CONVOLUTION_FWD_ALGO_IMPLICIT_GEMM";
+                auto convolution = static_cast<const ngraph::op::Convolution*>(node);
+                Strides window_dilation_strides = convolution->get_window_dilation_strides();
+                Strides window_movement_strides = convolution->get_window_movement_strides();
+                Strides data_dilation_strides = convolution->get_data_dilation_strides();
+                CoordinateDiff padding = convolution->get_padding_below();
+                CoordinateDiff padding_above = convolution->get_padding_above();
+
+                if (padding.size() > 3)
+                {
+                    throw std::runtime_error(node->get_name() +
+                                             "with more than 3D is not implemented.");
+                }
+                for (auto a : data_dilation_strides)
```

Comment:
nitpick: checks look the same for convolution forward and backward. I wonder if we could maybe refactor them into a helper?

Reply:
good point, we already had a discussion about this, will do it later.